src/request/derive-address/DerivedIdenticonSelector.js

```diff
@@ -0,0 +1,180 @@
+/* global Nimiq */
+/* global I18n */
+/* global Identicon */
+
+class DerivedIdenticonSelector extends Nimiq.Observable {
+    /**
+     * @param {HTMLElement} [$el]
+     */
+    constructor($el) {
+        super();
+
+        this._clearSelection = this._clearSelection.bind(this);
+        this._onSelectionConfirmed = this._onSelectionConfirmed.bind(this);
+
+        this.$el = DerivedIdenticonSelector._createElement($el);
+
+        /** @type {number} */
+        this._page = 1;
+
+        /** @type {string[]} */
+        this._pathsToDerive = [];
+
+        /** @type {{ [address: string]: { address: Nimiq.Address, keyPath: string } }} */
+        this._derivedAddresses = {};
+
+        /** @type {?string} */
+        this._selectedAddress = null;
+
+        this.$identicons = /** @type {HTMLElement} */ (this.$el.querySelector('.identicons'));
+        this.$confirmButton = /** @type {HTMLElement} */ (this.$el.querySelector('.backdrop button'));
+        this.$generateMoreButton = /** @type {HTMLElement} */ (this.$el.querySelector('.generate-more'));
+        this.$backdrop = /** @type {HTMLElement} */ (this.$el.querySelector('.backdrop'));
+
+        this.$generateMoreButton.addEventListener('click', this.nextPage.bind(this));
+        this.$backdrop.addEventListener('click', this._clearSelection);
+        this.$confirmButton.addEventListener('click', this._onSelectionConfirmed);
+    }
+
+    /**
+     * @param {HTMLElement} [$el]
+     *
+     * @returns {HTMLElement}
+     */
+    static _createElement($el) {
+        $el = $el || document.createElement('div');
+        $el.classList.add('identicon-selector');
+
+        $el.innerHTML = `
+            <div class="identicons">
+                <div class="loading center">
+                    <div class="loading-animation"></div>
+                    <h2 data-i18n="identicon-selector-loading">Mixing colors</h2>
+                </div>
+            </div>
+            <button class="generate-more">Generate new</button>
+
+            <div class="backdrop center">
+                <button data-i18n="identicon-selector-button-select">Select</button>
+                <a tabindex="0" class="secondary" data-i18n="identicon-selector-link-back">Back</a>
+            </div>`;
+
+        I18n.translateDom($el);
+        return $el;
+    }
+
+    /**
+     * @returns {HTMLElement}
+     */
+    getElement() {
+        return this.$el;
+    }
+
+    /**
+     * @param {Nimiq.ExtendedPrivateKey} masterKey
+     * @param {string[]} pathsToDerive
+     */
+    init(masterKey, pathsToDerive) {
+        /** @type {Nimiq.ExtendedPrivateKey} */
+        this._masterKey = masterKey;
+
+        /** @type {string[]} */
+        this._pathsToDerive = pathsToDerive;
+
+        this.generateIdenticons(this._masterKey, this._pathsToDerive, this._page);
+    }
+
+    nextPage() {
+        if (!this._masterKey) return;
+        this._page += 1;
+        if (this._page > 2) this._page = 1;
```

Comment:
The maximum of identicons to chose from is 20, as that is [the maximum allowed gap between used addresses for standardized automatic used-address-detection on wallet import (BIP44)](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki#Account_discovery). Since we display 7 identicons at a time, and 20 is not evenly devidable by 7, I thought it safer to go with only 14 choices. But I'll change it now to `floor(number of <indicesToDerive> / 7)`.